backend/backend/api/routes/room.py

```diff
@@ -0,0 +1,99 @@
+from functools import lru_cache
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, status, HTTPException
+
+from backend.common.room import Room
+from backend.common.player import Player
+from backend.api.models.room import RoomId, UpdateLeader
+from backend.api.models.player import PlayerId
+from backend.api.routes.player import get_players
+
+router = APIRouter(tags=["rooms"])
+
+
+@lru_cache
+def get_rooms() -> dict[str, Room]:
+    return {}
+
+
+@router.post("/rooms/", status_code=status.HTTP_201_CREATED,
+             response_model=RoomId)
+async def create_room(player_id: PlayerId,
+                      rooms: Annotated[dict[str, Room], Depends(get_rooms)],
+                      players: Annotated[dict[str, str], Depends(get_players)]):
+    room = Room()
+    player_name = players.get(player_id.player_id)
+
+    if not player_name:
+        raise HTTPException(detail=f"no user under given id {player_id.player_id}",
+                             status_code=status.HTTP_404_NOT_FOUND)
```

Comment:
Alright, that looks better,
Now, you can make it a dependency (the same way middleware would work in other frameworks)
so the actual route would only focus on the right logic to process

Reply:
done